astropy/io/fits/fitsrec.py

```diff
@@ -478,9 +478,22 @@ def from_columns(cls, columns, nrows=0, fill=False, character_as_bytes=False):
         return data
 
     def __repr__(self):
-        # Force use of the normal ndarray repr (rather than the new
-        # one added for recarray in Numpy 1.10) for backwards compat
-        return np.ndarray.__repr__(self)
+        repr_dtype = self.dtype
+        # FIXME: remove numpy.record in the dtype repr, as recarray?
+        # if repr_dtype.type is np.record:
+        #     repr_dtype = np.dtype((np.void, repr_dtype))
+        prefix = "FITS_rec("
+        fmt = "FITS_rec(%s,%sdtype=%s)"
+        # get data/shape string. logic taken from numeric.array_repr
+        if self.size > 0 or self.shape == (0,):
+            lst = np.array2string(
+                self, separator=", ", prefix=prefix, suffix=",", formatter=dict(int=str)
+            )
+        else:
+            # show zero-length shape unless it is (0,)
+            lst = "[], shape=%s" % (repr(self.shape),)  # noqa: UP031
```

Comment:
This should be equivalent and avoid a noqa, unless I'm missing something ?
```suggestion
            lst = f"[], shape={self.shape!r}"
```

Reply:
I copied this from Numpy so this is to stay close to the original.

Reply:
My suggestion of adding a copy on top would address this too, I think.